### ISSUE TITLE
ExtensionCallMarkLogic is synchronizing on calls to ML

### ISSUE THREAD

**Issue by rjrudin** [marklogic/nifi#48]
The block below is synchronizing on the call to ML, which creates a bottleneck when running the processor with multiple concurrent tasks. The synchronize only needs to be on the transfer/commit calls to the session. 

```
synchronized(resourceManager) {
    ServiceResultIterator resultIterator = resourceManager.callService(method, bytesHandle, requestParameters);
    while(resultIterator.hasNext()) {
        ServiceResult result = resultIterator.next();
        session.append(flowFile, out -> out.write(result.getContent(new BytesHandle()).get()));
    }
    session.transfer(flowFile, SUCCESS);
    session.commit();
}
```

**Comment by rjrudin** [comment]
Moved to https://github.com/marklogic-community/marklogic-nifi-incubator/issues/1

**Comment by rjrudin** [comment]
This was fixed in the incubator, and is thus in the 1.9.1.5 release
